### DISCUSSION TITLE
Improving working with top level statements programs by using `dotnet run algorithm1.cs`

### DISCUSSION THREAD

**Discussion by Swiftly1** [dotnet/csharplang#5429] [Language Ideas]
Let's say that I'm student having to do a few assignments:

I create project

`dotnet new console`

I write it in top-level statements

now I'd want to code next Exercise, 2.

I rename `Program.cs` to `Exercise1.cs` 

and as far as I see my options are either: refactor current code or create new console app

but what If it worked like in Python, IIRC

    python first_script.py
    Hello World!

that I could add just `Exercise2.cs` that would be in top-level statements too

and I could run `dotnet run Exercise2.cs`?

So I could have both files writen in top-level statement style and under same project

The problem is that now I'd receive error that `Only one compilation unit can have top-level statements.`

but with those changes it could say `Only one compilation unit can have top-level statements. Specify which top level program you want to run e.g "dotnet run Program2.cs"` 

To simplify things I believe it should only work with top-level statements and probably use `<ImplicitUsings>enable</ImplicitUsings>`

_____________

Additionally, what if Visual Studio's F5 button compiled `currently opened top-level statement` program?

I hope it makes sense to you, all I want to say that top-level statements are great, but if you want to write a few of them, then there's probably room for improvement when it comes to UX

**Comment by 333fred** [comment]
This is a language proposal, not a thing the SDK can do, so I moved the issue.

Personally, while I like the scenario, I think that it will be better solved by improved loose files support so you can just have your homework each in different files. We worry about the potential for confusion if there are multiple files in a single program with top level statements.

**Reply by 333fred** [reply]
See also https://github.com/dotnet/designs/pull/213.
